src/action/darwin/create_fstab_entry.rs

```diff
@@ -0,0 +1,237 @@
+
+        let mut file_contents = String::default();
+        file.read_to_string(&mut file_contents)
+            .await
+            .map_err(|e| ActionError::Read(fstab_path.to_owned(), e))?;
+
+        if let Some(start) = file_contents.rfind(fstab_entry.as_str()) {
+            let end = start + fstab_entry.len();
+            file_contents.replace_range(start..end, "")
+        }
+
+        file.seek(SeekFrom::Start(0))
+            .await
+            .map_err(|e| ActionError::Seek(fstab_path.to_owned(), e))?;
+        file.set_len(0)
+            .await
+            .map_err(|e| ActionError::Truncate(fstab_path.to_owned(), e))?;
+        file.write_all(file_contents.as_bytes())
+            .await
+            .map_err(|e| ActionError::Write(fstab_path.to_owned(), e))?;
+        file.flush()
+            .await
+            .map_err(|e| ActionError::Flush(fstab_path.to_owned(), e))?;
+
+        Ok(())
+    }
+}
+
+async fn get_uuid_for_label(apfs_volume_label: &str) -> Result<Option<Uuid>, ActionError> {
+    let output = execute_command(
+        Command::new("/usr/sbin/diskutil")
+            .process_group(0)
+            .arg("info")
+            .arg(apfs_volume_label)
+            .stdin(std::process::Stdio::null())
+            .stdout(std::process::Stdio::piped()),
+    )
+    .await
+    .map_err(|e| ActionError::Command(e))?;
+
+    let stdout = String::from_utf8(output.stdout)?;
+
+    let mut found = None;
+    for line in stdout.lines() {
+        let prefix = "Volume UUID:";
+        let trimmed = line.trim();
+        if let Some(index) = trimmed.find(prefix) {
+            let maybe_uuid = trimmed[(index + prefix.len())..].trim();
+            let uuid = Uuid::parse_str(maybe_uuid).map_err(|err| {
+                ActionError::Custom(Box::new(CreateFstabEntryError::Uuid(
+                    maybe_uuid.to_string(),
+                    err,
+                )))
+            })?;
+
+            found = Some(uuid);
+            break;
+        }
+    }
+
+    Ok(found)
+}
+
+fn fstab_prelude_comment(apfs_volume_label: &str) -> String {
+    format!("# nix-installer created volume labelled `{apfs_volume_label}`")
+}
+
+fn fstab_entry(uuid: &Uuid, apfs_volume_label: &str) -> String {
+    let prelude_comment = fstab_prelude_comment(apfs_volume_label);
+    format!(
+        "\
+        {prelude_comment}\n\
+        UUID=\"{uuid}\" /nix apfs rw,noauto,nobrowse,suid,owners\n\
+        "
+    )
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum CreateFstabEntryError {
+    #[error("UUID error: {0}")]
+    Uuid(String, #[source] uuid::Error),
+    #[error("No volume labelled `{0}` present, cannot get UUID to add to /etc/fstab")]
+    NoVolume(String),
+    #[error("An `/etc/fstab` entry for the volume labelled `{0}` already exists, if a Nix Store already exists it may need to be removed with `diskutil apfs deleteVolume \"{0}\"), this entry should be removed from `/etc/fstab`")]
```

Comment:
```suggestion
    #[error("An `/etc/fstab` entry for the volume labelled `{0}` already exists. If a Nix Store already exists it may need to be deleted with `diskutil apfs deleteVolume \"{0}\" and should be removed from `/etc/fstab`")]
```

or something like that.